clvm-traits/src/curried_args.rs

```diff
@@ -0,0 +1,86 @@
+use clvmr::{
+    allocator::{NodePtr, SExp},
+    Allocator,
+};
+
+use crate::{
+    clvm_list, clvm_quote, destructure_list, destructure_quote, match_list, match_quote, Error,
+    FromClvm, MatchByte, Result, ToClvm,
+};
+
+#[derive(Debug, Clone)]
+pub struct CurriedArgs(pub Vec<NodePtr>);
```

Comment:
don't you need a way to get the inner puzzle as well?
in the future, it would be nice to have a way to get the curried args into an array as well, to avoid the heap allocation, when you know what the puzzle is supposed to be